### PR TITLE
ports mesh component to latest API (#6455)

### PR DESCRIPTION
**Pull request by otsakir**
Ports mesh component to latest component api.

Mesh resource porting will be done in a separate PR.

### CODE REVIEW

File: engine/engine/src/test/wscript [hunk 1]
```diff
@@ -57,7 +57,7 @@ def build(bld):
 
     resource_type_symbols = ['ResourceTypeGameObject', 'ResourceTypeCollection', 'ResourceTypeScript', 'ResourceTypeLua',
                              'ResourceTypeAnim', 'ResourceTypeAnimationSet', 'ResourceTypeGui', 'ResourceTypeGuiScript']
-    component_type_symbols = ['ComponentTypeAnim', 'ComponentTypeScript', 'ComponentTypeGui']
+    component_type_symbols = ['ComponentTypeAnim', 'ComponentTypeScript', 'ComponentTypeGui', 'ComponentTypeMesh']
```
Comment (JCash) [inline engine/engine/src/test/wscript:60]: The "build_input.yml" file is missing in this PR?

File: engine/gamesys/src/gamesys/components/comp_mesh.cpp [hunk 2]
```diff
@@ -1037,4 +1037,59 @@ namespace dmGameSystem
         pit->m_FnIterateNext = CompMeshIterPropertiesGetNext;
     }
 
+    static dmGameObject::Result CompMeshTypeCreate(const dmGameObject::ComponentTypeCreateCtx* ctx, dmGameObject::ComponentType* type)
+    {
+        MeshContext* mesh_context = new MeshContext;
```
Comment (JCash) [inline engine/gamesys/src/gamesys/components/comp_mesh.cpp:1042]: The mesh context can now be moved to within this .cpp file.

File: engine/gamesys/src/gamesys/gamesys.cpp [hunk 3]
```diff
@@ -243,13 +243,7 @@ namespace dmGameSystem
...
                 0);
 
-        REGISTER_COMPONENT_TYPE("meshc", 725, mesh_context,
-                CompMeshNewWorld, CompMeshDeleteWorld,
-                CompMeshCreate, CompMeshDestroy, 0, 0, CompMeshAddToUpdate, 0,
-                CompMeshUpdate, 0, CompMeshRender, 0, CompMeshOnMessage, 0,
-                0, CompMeshGetProperty, CompMeshSetProperty,
-                0, CompMeshIterProperties,
-                0);
+        // prio: 725  comp_mesh.cpp
```
Comment (JCash) [inline engine/gamesys/src/gamesys/gamesys.cpp:246]: Remove the mesh context from the function.

**Review by JCash — CHANGES_REQUESTED** [review]

File: engine/gamesys/src/gamesys/components/comp_mesh.cpp [hunk 4]
```diff
@@ -1037,4 +1037,59 @@ namespace dmGameSystem
...
+        mesh_context->m_MaxMeshCount = dmConfigFile::GetInt(ctx->m_Config, "mesh.max_count", 128); // TODO - also initialized in engine.cpp:1106. Why ?
+
+//        int32_t max_gui_count = dmConfigFile::GetInt(ctx->m_Config, "gui.max_instance_count", 128);
+//        gui_context->m_Worlds.SetCapacity(max_gui_count);
+//        dmGui::InitializeScript(gui_context->m_ScriptContext);
+
+        ComponentTypeSetPrio(type, 725);
+
+        ComponentTypeSetContext(type, mesh_context);
+//        ComponentTypeSetHasUserData(type, true);      // TODO  ???
```
Comment (JCash) [inline engine/gamesys/src/gamesys/components/comp_mesh.cpp:1054]: If you see the original code, it sets it to true.

Comment (otsakir) [reply engine/gamesys/src/gamesys/components/comp_mesh.cpp:1054]: Yeah, i just found out. Also, there is a memset(0) that zeros out all the "type" struct.

File: engine/gamesys/src/gamesys/gamesys.cpp (hunk 3 above)
Comment (otsakir) [reply engine/gamesys/src/gamesys/gamesys.cpp:246]: Ok. 

There also the m_MeshContext property in struct Engine. I searched for references throughout the codebase. It doesn't seem to be used. I removed that too. It compiles and runs ok.

File: engine/gamesys/src/gamesys/components/comp_mesh.cpp (hunk 2 above)
Comment (otsakir) [reply engine/gamesys/src/gamesys/components/comp_mesh.cpp:1042]: Ok. Nice...

File: engine/engine/src/test/wscript (hunk 1 above)
Comment (JCash) [reply engine/engine/src/test/wscript:60]: We are talking about the C++ symbols, in your case `ComponentTypeMesh`.
It needs to be specified for the engine to be able to link with it.

See "symbols" at the top of the file, search for `ComponentTypeGui` as an example.

File: engine/gamesys/src/gamesys/gamesys.cpp (hunk 3 above)
Comment (JCash) [reply engine/gamesys/src/gamesys/gamesys.cpp:246]: yes

File: engine/engine/src/test/wscript (hunk 1 above)
Comment (otsakir) [reply engine/engine/src/test/wscript:60]: Ok. I've added it. 

Note, it built and ran a test project with mesh properly even without this change.

Comment (JCash) [reply engine/engine/src/test/wscript:60]: "Without this change"
Are you talking about build_input.yml?
This config file is used when you build a native extension (you have to use your sha1 version of the sdk)
Did you build a project with native extensions? With your sha1?

**Review by JCash — APPROVED** [review]

**Pull request merged**